pkg/sql/conn_executor.go

```diff
@@ -736,7 +736,13 @@ func (s *Server) getScrubbedStmtStats(
 
 		// Possibly scrub the app name.
 		if !strings.HasPrefix(stat.Key.App, catconstants.ReportableAppNamePrefix) {
-			stat.Key.App = HashForReporting(salt, stat.Key.App)
+			if !strings.HasPrefix(stat.Key.App, catconstants.DelegatedAppNamePrefix) {
+				stat.Key.App = HashForReporting(salt, stat.Key.App)
+			} else if !strings.HasPrefix(stat.Key.App,
+				catconstants.DelegatedAppNamePrefix+catconstants.ReportableAppNamePrefix) {
+				stat.Key.App = catconstants.DelegatedAppNamePrefix + HashForReporting(
+					salt, stat.Key.App)
+			}
```

Comment:
I think this block would benefit either from a detailed comment or extraction into a function to explain what's going on and allow for some nice unit testing with examples.